src/main/java/graphql/language/ChildrenContainer.java

```diff
@@ -0,0 +1,77 @@
+package graphql.language;
+
+import graphql.Assert;
+import graphql.PublicApi;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+@PublicApi
+public class ChildrenContainer {
+
+    private final Map<String, List<Node>> children = new LinkedHashMap<>();
+
+    private ChildrenContainer(Map<String, List<Node>> children) {
+        this.children.putAll(Assert.assertNotNull(children));
+    }
+
+    public <T extends Node> List<T> getList(String key) {
```

Comment:
`getChildrenWithName` or just `getChildren` is a better namer I think.  You know its a List from the return type

Reply:
still stand by this comment

Reply:
will change